ansible/roles/csm.storage.smartmon/tasks/main.yml

```diff
@@ -34,10 +34,16 @@
     path: "{{ ceph_admin_keyring_path }}"
   register: admin_keyring
 
-- name: Redeploy node-exporter
+- name: Get Ceph version
   when:
     - admin_keyring.stat.exists
-  command: "cephadm shell --mount /etc/cray/ceph/ -- ceph orch apply -i /mnt/node-exporter.yml"
+  shell: "ceph version -f json | jq .version | awk '{ print $3 }'"
+  register: ceph_version
+
+- name: Redeploy node-exporter
+  when:
+    - (ceph_version.rc is defined) and (ceph_version.rc == 0)
+  command: "cephadm --image {{ ceph_container_path }}:v{{ ceph_version.stdout }} shell --mount /etc/cray/ceph/ -- ceph orch apply -i /mnt/node-exporter.yml"
```

Comment:
I do wonder about using the 'v' in the version whether that's a little dangerous or not. It's one of those things that easy to forget. It's hard-coded here. As long as the version is always vX.Y.Z we're okay. I'm sure you didn't choose the 'v' prefix for versioning.

Reply:
The container version does have a 'v' in front and the value received from the command that gets the ceph version does not contain a 'v'. So I believe it is okay the way it is written now, but I understand the concern!